packages/java-parser/src/productions/classes.js

```diff
@@ -795,6 +795,8 @@ function defineRules($, t) {
       let nextNextTokenType = this.LA(2).tokenType;
 
       switch (nextTokenType) {
+        case t.Record:
+          return classBodyTypes.classDeclaration;
```

Comment:
I wonder if this check should not be placed below with:

```js
if (
        tokenMatcher(nextTokenType, t.Class) ||
        tokenMatcher(nextTokenType, t.Enum)
      ) {
        return classBodyTypes.classDeclaration;
      }
```

I suspect it won't work if you add annotation or class modifiers otherwise. Could you add a unit test with modifiers so that we can check and prevent any regression ?

Reply:
Nice one, you are right
I added annotation and modifiers and does works well